### ISSUE TITLE
FindPython modules require CMake 3.7+

### ISSUE THREAD

**Issue by PeterBowman** [robotology/ycm#248]
Any attempt to `find_package(Python)` on CMake <3.7 will result in:

```
CMake Error at /usr/local/share/YCM/cmake-next/Modules/FindPython/Support.cmake:8 (cmake_policy):
  An attempt was made to set the policy version of CMake to "3.7" which is
  greater than this version of CMake.  This is not allowed because the
  greater version may have new policies not known to this CMake.  You may
  need a newer CMake version to build this project.
```

Cause: https://github.com/robotology/ycm/commit/6bd35ad8f618f0da9df4cb982a25ed349fa55bc8 + `cmake_policy (VERSION 3.7)` ([ref](https://github.com/Kitware/CMake/blob/v3.12.4/Modules/FindPython/Support.cmake#L8)).

**Comment by drdanz** [comment]
In #251 I simply replaced

```diff
-cmake_policy (VERSION 3.7)
+if (NOT CMAKE_VERSION VERSION_LESS 3.7)
+  cmake_policy (VERSION 3.7)
+endif ()
```

I didn't get any error with CMake 3.5, but I cannot guarantee that this won't have any side effects.
@PeterBowman do you have any project where you can try this?

**Comment by PeterBowman** [comment]
@drdanz it works for me with CMake 3.5, too, thanks!

**Comment by drdanz** [comment]
Perfect, thanks! Merged #251
